lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
It makes a hell lot of difference. To give you example, with 50 million
dimensional vectors and with 1.4 GB of data in memory per mapper, the peak
memory usage of evaluator is 8GB without GC.Collect and 2.8 GB with
GC.Collect. Let me give you exact time measurements.

One way to do is following: We can give user an option for this. i.e. in
IMRU give a configurable parameters say Boolean called MemoryEfficient that
allows user to trade off memory efficiency with overhead of GC.


Dhruv

On Mon, Sep 28, 2015 at 3:58 PM, Markus Weimer <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs
> <https://github.com/apache/incubator-reef/pull/527#discussion_r40616558>:
>
> >              {
> > +                GC.Collect();
>
> lemme do the measurements
>
> Alternatively, we could merge this without the GC calls. Have you run out
> of memory without them? Just using lots of memory isn't bad, as long as we
> stay within the container boundaries.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/527/files#r40616558>.
>